be1-go/channel/lao/lao.go

```diff
@@ -768,3 +775,28 @@ func (c *Channel) extractLaoID() string {
 func (r *rollCall) checkPrevID(prevID []byte) bool {
 	return string(prevID) == r.id
 }
+
+// checkIsFromOrganizer is a helper method which validates that the message's
+// sender is the organizer. Return an error if it failed or if it's false,
+// return nil if it was from the organizer.
+func (c *Channel) checkIsFromOrganizer(msg message.Message) error {
+	senderBuf, err := base64.URLEncoding.DecodeString(msg.Sender)
+	if err != nil {
+		return xerrors.Errorf(keyDecodeError, err)
```

Comment:
Is there a reason why this is built using `xerrors.Errorf` not as an `amswer.NewErrorf` like the other errors returned ?

Reply:
It was because this function was created using some existing code.
After reading answer/error.go, I just found that there are some functions with the error code already inside,
such as `answer.NewInvalidMessageFieldError` and `answer.NewAccessDeniedError`.